File: base/src/main/java/com/zhangteng/base/base/TitlebarActivity.java
package com.zhangteng.base.base;

import android.view.View;

import com.zhangteng.base.R;
import com.zhangteng.base.widget.CommonTitleBar;


/**
 * Created by swing on 2017/11/23.
 */

public abstract class TitlebarActivity extends BaseActivity {
    protected CommonTitleBar mTitlebar;

    @Override
    protected void initView() {
        mTitlebar = findViewById(R.id.titlebar);
        if (mTitlebar == null) {
            throw new IllegalStateException(
                    "The subclass of TitlebarActivity must contain a titlebar.");
        }
        mTitlebar.setListener(new CommonTitleBar.OnTitleBarListener() {
            @Override
            public void onClicked(View v, int action, String extra) {
                if (action == CommonTitleBar.ACTION_LEFT_BUTTON || action == CommonTitleBar.ACTION_LEFT_TEXT) {
                    finish();
                }
            }
        });
    }
}

File: base/src/main/java/com/zhangteng/base/base/TitlebarActivity.kt
package com.zhangteng.base.base

import android.view.View
import com.zhangteng.base.R
import com.zhangteng.base.widget.CommonTitleBar
import com.zhangteng.base.widget.CommonTitleBar.OnTitleBarListener

/**
 * Created by swing on 2017/11/23.
 */
abstract class TitlebarActivity : BaseActivity() {
    protected var mTitlebar: CommonTitleBar? = null
    override fun initView() {
        mTitlebar = findViewById(R.id.titlebar)
        checkNotNull(mTitlebar) { "The subclass of TitlebarActivity must contain a titlebar." }
        mTitlebar?.setListener(object : OnTitleBarListener {
            override fun onClicked(v: View?, action: Int, extra: String?) {
                if (action == CommonTitleBar.ACTION_LEFT_BUTTON || action == CommonTitleBar.ACTION_LEFT_TEXT) {
                    finish()
                }
            }
        })
    }
}